src/main/java/ru/hogwarts/school/repository/FacultyRepository.java

```diff
@@ -3,5 +3,11 @@
 import org.springframework.data.jpa.repository.JpaRepository;
 import ru.hogwarts.school.model.Faculty;
 
+import java.util.Set;
+
 public interface FacultyRepository extends JpaRepository<Faculty, Long> {
+
+    Set<Faculty> getByColor(String color);
+
+    Set<Faculty> getByColorOrNameIgnoreCase(String color, String name);
```

Comment:
```suggestion
    Set<Faculty> getByColorIgnoreCaseOrNameIgnoreCase(String color, String name);
```
иначе только по одному полю игнорируем